tests/import/proto/bar/bar_host.pb.go

```diff
@@ -12,47 +12,42 @@ import (
 	context "context"
 	errors "errors"
 	fmt "fmt"
+	options "github.com/knqyf263/go-plugin/options"
 	wazero "github.com/tetratelabs/wazero"
 	api "github.com/tetratelabs/wazero/api"
 	wasi_snapshot_preview1 "github.com/tetratelabs/wazero/imports/wasi_snapshot_preview1"
 	sys "github.com/tetratelabs/wazero/sys"
-	io "io"
-	fs "io/fs"
 	os "os"
 )
 
 const BarPluginAPIVersion = 1
 
-type BarPluginOption struct {
-	Stdout io.Writer
-	Stderr io.Writer
-	FS     fs.FS
-}
+type BarPluginOption func(plugin *BarPlugin)
 
 type BarPlugin struct {
-	cache  wazero.CompilationCache
-	config wazero.ModuleConfig
+	newRuntime   func(context.Context) (wazero.Runtime, error)
+	moduleConfig wazero.ModuleConfig
+	runtimes     []wazero.Runtime
 }
 
-func NewBarPlugin(ctx context.Context, opt BarPluginOption) (*BarPlugin, error) {
-
-	// Create a new WebAssembly CompilationCache.
-	cache := wazero.NewCompilationCache()
+func NewBarPlugin(ctx context.Context, opts ...options.WazeroConfigOption) (*BarPlugin, error) {
+	o := options.NewWazeroConfig()
 
-	// Combine the above into our baseline config, overriding defaults.
-	config := wazero.NewModuleConfig().
-		// By default, I/O streams are discarded and there's no file system.
-		WithStdout(opt.Stdout).WithStderr(opt.Stderr).WithFS(opt.FS)
+	for _, opt := range opts {
+		opt(o)
+	}
 
 	return &BarPlugin{
-		cache:  cache,
-		config: config,
+		newRuntime:   o.NewRuntime,
+		moduleConfig: o.ModuleConfig,
 	}, nil
 }
 
 func (p *BarPlugin) Close(ctx context.Context) (err error) {
-	if c := p.cache; c != nil {
-		err = c.Close(ctx)
+	for i := range p.runtimes {
```

Comment:
I now understand the tension here about Close, as the BarPlugin is really a BarPluginFactory, which makes BarPlugin on Load... caching the runtimes here does make sure they close, but also much cleaner if the BarPlugin themselves can close.

Sorry it took code to get to the point of understanding this. I think that we shouldn't be caching runtimes here, rather plugins, and close call close on them. This will allow a user to cast into `func Close(ctx context.Context) error` and cleanup their plugins when they are done with the corresponding runtime. you could also define io.Closer which dispatches to those using `context.Background()`. This way when someone declares io.Closer a sensible implementation is there already by default.

@knqyf263 on this point escalating to your thoughts as I think this is the better plan but could use another opinion!